### DISCUSSION TITLE
Monorepo - separate codespace for each app

### DISCUSSION THREAD

**Discussion by irwinbraganza** [community/community#16904] [Codespaces] [answered]
<img width="444" alt="Screenshot 2022-05-19 at 11 14 26" src="https://user-images.githubusercontent.com/8552879/169258520-e11d4bc6-1b05-4542-9917-8e8fad0d6692.png">

in the above lerna mono-repo setup, can we have
1.  the main repository containing the packages ?
and
2. the Apps (App1, App2, App3) each App having its own codespace?

primary aim would be: we want a setup to easily use shared code + we want each app to have its own owner/maintainer

**Comment by dmilburn** [comment, marked as answer]
Hi from the Codespaces team! Hmm it sounds like maybe what you're looking for is each app having its own devcontainer set up. See: https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#about-dev-containers 

> When multiple configurations are available, users can choose their preferred configuration when they create a codespace. This is particularly useful for large repositories that contain source code in different programming languages or for different projects. You can create a choice of configurations that allow different teams to work in a codespace that's set up appropriately for the work they are doing.
